scripts/dispatch_table_helper_generator.py

```diff
@@ -18,219 +18,84 @@
-        preamble += '#include <string.h>\n'
-        preamble += '#include "loader/generated/vk_layer_dispatch_table.h"\n'
-
-        write(copyright, file=self.outFile)
-        write(preamble, file=self.outFile)
-    #
-    # Write generate and write dispatch tables to output file
-    def endFile(self):
-        device_table = ''
-        instance_table = ''
-
-        device_table += self.OutputDispatchTableHelper('device')
-        instance_table += self.OutputDispatchTableHelper('instance')
-
-        for stub in self.dev_ext_stub_list:
-            write(stub, file=self.outFile)
-        write("\n\n", file=self.outFile)
-        write(device_table, file=self.outFile);
-        write("\n", file=self.outFile)
-        write(instance_table, file=self.outFile);
+    def generate(self):
+        out = []
 
-        # Finish processing in superclass
-        OutputGenerator.endFile(self)
-    #
-    # Processing at beginning of each feature or extension
-    def beginFeature(self, interface, emit):
-        OutputGenerator.beginFeature(self, interface, emit)
-        self.featureExtraProtect = GetFeatureProtect(interface)
-        self.extension_type = interface.get('type')
-
-    #
-    # Process commands, adding to appropriate dispatch tables
-    def genCmd(self, cmdinfo, name, alias):
-        OutputGenerator.genCmd(self, cmdinfo, name, alias)
-
-        avoid_entries = ['vkCreateInstance',
-                         'vkCreateDevice']
-        # Get first param type
-        params = cmdinfo.elem.findall('param')
-        info = self.getTypeNameTuple(params[0])
-
-        if name not in avoid_entries:
-            self.AddCommandToDispatchList(name, info[0], self.featureExtraProtect, cmdinfo)
-
-    #
-    # Determine if this API should be ignored or added to the instance or device dispatch table
-    def AddCommandToDispatchList(self, name, handle_type, protect, cmdinfo):
-        handle = self.registry.tree.find("types/type/[name='" + handle_type + "'][@category='handle']")
-        if handle is None:
-            return
-        if handle_type != 'VkInstance' and handle_type != 'VkPhysicalDevice' and name != 'vkGetInstanceProcAddr':
-            self.device_dispatch_list.append((name, self.featureExtraProtect))
-        else:
-            self.instance_dispatch_list.append((name, self.featureExtraProtect))
-        return
-    #
-    # Retrieve the type and name for a parameter
-    def getTypeNameTuple(self, param):
-        type = ''
-        name = ''
-        for elem in param:
-            if elem.tag == 'type':
-                type = noneStr(elem.text)
-            elif elem.tag == 'name':
-                name = noneStr(elem.text)
-        return (type, name)
-    #
-    # Create a dispatch table from the appropriate list and return it as a string
-    def OutputDispatchTableHelper(self, table_type):
-        entries = []
-        table = ''
+        out.append('#pragma once\n')
```

Comment:
any reason to not do a `'''` multi-block here so it is only a single append (like https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/main/scripts/generators/api_version_generator.py#L89)